src/components/PerformanceOptimizer.tsx

```
@@ -0,0 +1,100 @@
+"use client";
+
+import { ReactNode, useEffect, useRef, useState } from "react";
+import { motion } from "framer-motion";
+
+interface PerformanceOptimizerProps {
+  children: ReactNode;
+  preset?: "fade-in" | "slide-up" | "slide-down" | "zoom-in" | "none";
+  customAnimation?: {
+    initial: any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)